parse-zoneinfo/src/line.rs

```diff
@@ -810,6 +777,191 @@ pub struct ZoneInfo<'a> {
     pub time: Option<ChangeTime>,
 }
 
+impl<'a> ZoneInfo<'a> {
+    fn from_iter(iter: impl Iterator<Item = &'a str>) -> Result<Self, Error> {
+        let mut state = ZoneInfoState::Start;
+        for part in iter {
+            state = match (state, part) {
+                (st, _) if part.starts_with('#') => {
```

Comment:
In theory a comment is allowed to come straight after a field, without whitespace in between.